department-of-veterans-affairs/caseflow
# New appeal factory trait

Connects #10886 

### Description
Adds new Appeal factory trait for a realistic Appeal at Attorney Decision Drafting.

Reorders the Appeal factory.

### Acceptance Criteria

### Testing Plan
1. Load the dev rails console
1. Create an Appeal with the trait:
    ```
   judge = FactoryBot.create(:user, css_id: "test")
   FactoryBot.create(:staff, :judge_role, sdomainid: judge.css_id)
   jt = JudgeTeam.create_for_judge(judge)
   atty = FactoryBot.create(:user, css_id: "test_atty")
   FactoryBot.create(:staff, :attorney_role, user: atty)
   jt.add_user(atty)
   appeal = FactoryBot.create(
      :appeal,
      :at_attorney_drafting,
      docket_type: Constants.AMA_DOCKETS.direct_review,
      associated_judge: judge,
      associated_attorney: atty,
      veteran: FactoryBot.create(:veteran, file_number: "100009659"),
      request_issues: FactoryBot.build_list(
        :request_issue, 1,
        contested_issue_description: "Tinnitus"
      )
    )
    ```
1. Confirm appeal has a proper tree with statuses
    ```
    > puts appeal.structure_render(:id, :status, :assigned_to_id, :assigned_by_id)
    Appeal 283 [id, status, assigned_to_id, assigned_by_id]
    └── RootTask 1160, on_hold, 4, (assigned_by_id)
        ├── DistributionTask 1161, completed, 4, (assigned_by_id)
        ├── JudgeAssignTask 1162, completed, 214, (assigned_by_id)
        └── JudgeDecisionReviewTask 1163, on_hold, 214, (assigned_by_id)
            └── AttorneyTask 1164, assigned, 215, 214

    ```
### User Facing Changes
None
